File: 45_jump_game_2/jump.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        k = j = c = 0

        for i in range(len(nums) - 1):
            j = max(j, nums[i] + i)
            if i == k:
                c += 1
                k = j
        
        return c

File: 45_jump_game_2/jump.java
class Solution {
    public int jump(int[] nums) {
        int reach = 0;
        int jumped = 0;
        int count = 0;
        int i = 0;

        while (i < nums.length - 1) {
            reach = Math.max(reach, nums[i] + i);

            if (i == jumped) {
                jumped = reach;
                count++;
            }
            i++;
        }

        return count;
    }
}